crates/ruma-common/src/events/room/member.rs

```diff
@@ -269,6 +288,15 @@ impl SignedContent {
     }
 }
 
+/// The details of a member event required to calculate a [`MembershipChange`].
+pub struct MembershipDetails<'a> {
```

Comment:
Couldn't this be `MembershipProfile`?

Reply:
Maybe `MemberProfileBits` or sth. like that? It's really not meant to be useful for anything other than that method so I didn't put too mucb thought in it.

Reply:
I think that ProfileBits and/or without Details would be even worse, i thought MembershipProfile as "profile" is mentioned often in this area in the spec.

Reply:
But it's intentionally a type that just holds the necessary data for change calculation, whatever that is (doesn't have to be profile-related at all).